pulp_container/app/registry_api.py

```diff
@@ -449,6 +452,19 @@ class CatalogView(ContainerRegistryApiMixin, ListAPIView):
     queryset = models.ContainerDistribution.objects.all().only("base_path")
     serializer_class = ContainerCatalogSerializer
     pagination_class = ContainerCatalogPagination
+    access_policy_class = RegistryAccessPolicy()
+
+    def get_queryset(self, *args, **kwargs):
+        """Filter the queryset based on assigned permissions."""
+        queryset = super().get_queryset()
+
+        accessible_repo_pks = []
+        for obj in queryset:
+            view = FakeView("pull", lambda: obj)
+            if self.access_policy_class.has_permission(self.request, view):
+                accessible_repo_pks.append(obj.pk)
+
+        return queryset.filter(pk__in=accessible_repo_pks)
```

Comment:
you probably want to order the results alphabetically or by date, so your results are deterministic. This is also can be the reason of the failing tests, when 2 lists are compared items seem to have different sequence

Reply:
results get ordered in the pagination class, no need to order